src/Compilers/CSharp/Test/Emit/CodeGen/CodeGenAwaitForeachTests.cs

```diff
@@ -4796,6 +4885,51 @@ public async Task<int> DisposeAsync()
             CompileAndVerify(comp, expectedOutput: "MoveNextAsync DisposeAsync Done");
         }
 
+        [Fact]
+        public void PatternBasedDisposal_WithOptionalParameter()
+        {
+            string source = @"
+using System.Threading.Tasks;
+class C
+{
+    public static async Task Main()
+    {
+        await foreach (var i in new C())
+        {
+        }
+        System.Console.Write(""Done"");
+    }
+    public Enumerator GetAsyncEnumerator()
+    {
+        return new Enumerator();
+    }
+    public sealed class Enumerator
+    {
+        public async Task<bool> MoveNextAsync()
```

Comment:
>() [](start = 45, length = 2)

Can this method have optional parameters? Do we have tests covering this scenario?

Reply:
See `TestMoveNextAsync_WithOptionalParameter` and `TestMoveNextAsync_WithParamsParameter` in this file.

---
In reply to: [550213568](https://github.com/dotnet/roslyn/pull/50105#discussion_r550213568) [](ancestors = 550213568)